File: mars_logging_android/android/src/main/kotlin/dev/zeekr/mars_logging_android/XLogImpl.kt
package dev.zeekr.mars_logging_android

class XLogImpl : XLogHostApi {
    override fun open(
        mode: AppenderModeApi,
        level: XLogLevelApi,
        logsDir: String,
        cacheDir: String,
        cacheDays: Long,
        namePrefix: String,
        compressMode: CompressModeApi,
        compressLevel: CompressLevelApi,
        pubKey: String,
        useConsole: Boolean,
        maxFileSize: Long,
        maxAliveDuration: Long
    ) {
        XLog.open(
            mode.impl,
            level.impl,
            logsDir,
            cacheDir,
            cacheDays.toInt(),
            namePrefix,
            compressMode.impl,
            compressLevel.impl,
            pubKey,
            useConsole,
            maxFileSize,
            maxAliveDuration,
        )
    }

    override fun flush(isSync: Boolean) {
        XLog.flush(isSync)
    }

    override fun close() {
        XLog.close()
    }

    override fun verbose(tag: String, message: String) {
        XLog.verbose(tag, message)
    }

    override fun debug(tag: String, message: String) {
        XLog.debug(tag, message)
    }

    override fun info(tag: String, message: String) {
        XLog.info(tag, message)
    }

    override fun warning(tag: String, message: String) {
        XLog.warning(tag, message)
    }

    override fun error(tag: String, message: String) {
        XLog.error(tag, message)
    }

    override fun fatal(tag: String, message: String) {
        XLog.fatal(tag, message)
    }
}

val AppenderModeApi.impl: AppenderMode
    get() = when (this) {
        AppenderModeApi.ASYNC -> AppenderMode.ASYNC
        AppenderModeApi.SYNC -> AppenderMode.SYNC
    }

val XLogLevelApi.impl: XLogLevel
    get() = when (this) {
        XLogLevelApi.ALL -> XLogLevel.ALL
        XLogLevelApi.VERBOSE -> XLogLevel.VERBOSE
        XLogLevelApi.DEBUG -> XLogLevel.DEBUG
        XLogLevelApi.INFO -> XLogLevel.INFO
        XLogLevelApi.WARNING -> XLogLevel.WARNING
        XLogLevelApi.ERROR -> XLogLevel.ERROR
        XLogLevelApi.FATAL -> XLogLevel.FATAL
        XLogLevelApi.NONE -> XLogLevel.NONE
    }

val CompressModeApi.impl: CompressMode
    get() = when (this) {
        CompressModeApi.ZLIB -> CompressMode.ZLIB
        CompressModeApi.ZSTD -> CompressMode.ZSTD
    }

val CompressLevelApi.impl: CompressLevel
    get() = when (this) {
        CompressLevelApi.LEVEL1 -> CompressLevel.LEVEL1
        CompressLevelApi.LEVEL2 -> CompressLevel.LEVEL2
        CompressLevelApi.LEVEL3 -> CompressLevel.LEVEL3
        CompressLevelApi.LEVEL4 -> CompressLevel.LEVEL4
        CompressLevelApi.LEVEL5 -> CompressLevel.LEVEL5
        CompressLevelApi.LEVEL6 -> CompressLevel.LEVEL6
        CompressLevelApi.LEVEL7 -> CompressLevel.LEVEL7
        CompressLevelApi.LEVEL8 -> CompressLevel.LEVEL8
        CompressLevelApi.LEVEL9 -> CompressLevel.LEVEL9
    }

File: mars_logging_darwin/darwin/Classes/XLogImpl.swift
//
//  XLogImpl.swift
//  mars_logging
//
//  Created by 闫守旺 on 2025/3/19.
//

import Foundation

class XLogImpl: XLogHostApi {
    func open(mode: AppenderModeApi, level: XLogLevelApi, logsDir: String, cacheDir: String, cacheDays: Int64, namePrefix: String, compressMode: CompressModeApi, compressLevel: CompressLevelApi, pubKey: String, useConsole: Bool, maxFileSize: Int64, maxAliveDuration: Int64) throws {
        XLog.open(mode.impl, level: level.impl, logsDir: logsDir, cacheDir: cacheDir, cacheDays: Int32(cacheDays), namePrefix: namePrefix, compressMode: compressMode.impl, compressLevel: compressLevel.impl, pubKey: pubKey, useConsole: useConsole, maxFileSize: UInt64(maxFileSize), maxAliveDuration: Int(maxAliveDuration))
    }
    
    func flush(isSync: Bool) throws {
        XLog.flush(isSync)
    }
    
    func close() throws {
        XLog.close()
    }
    
    func verbose(tag: String, message: String) throws {
        XLog.verbose(tag, message: message)
    }
    
    func debug(tag: String, message: String) throws {
        XLog.debug(tag, message: message)
    }
    
    func info(tag: String, message: String) throws {
        XLog.info(tag, message: message)
    }
    
    func warning(tag: String, message: String) throws {
        XLog.warning(tag, message: message)
    }
    
    func error(tag: String, message: String) throws {
        XLog.error(tag, message: message)
    }
    
    func fatal(tag: String, message: String) throws {
        XLog.fatal(tag, message: message)
    }
}

extension AppenderModeApi {
    var impl: AppenderMode {
        switch self {
        case .async:
            return .async
        case .sync:
            return .sync
        }
    }
}

extension XLogLevelApi {
    var impl: XLogLevel {
        switch self {
        case .all:
            return .all
        case .verbose:
            return .verbose
        case .debug:
            return .debug
        case .info:
            return .info
        case .warning:
            return .warning
        case .error:
            return .error
        case .fatal:
            return .fatal
        case .none:
            return .none
        }
    }
}

extension CompressModeApi {
    var impl: CompressMode {
        switch self {
        case .zlib:
            return .zlib
        case .zstd:
            return .zstd
        }
    }
    
}

extension CompressLevelApi {
    var impl: CompressLevel {
        switch self {
        case .level1:
            return .level1
        case .level2:
            return .level2
        case .level3:
            return .level3
        case .level4:
            return .level4
        case .level5:
            return .level5
        case .level6:
            return .level6
        case .level7:
            return .level7
        case .level8:
            return .level8
        case .level9:
            return .level9
        }
    }
}
